src/GameLogic/TurnRunner.ts

```diff
@@ -41,6 +42,31 @@ export class TurnRunner {
 						// @todo When campaignieren are implementieren, implement a way to return that this level has been completed
 					}
 				}
+
+				if (level.tilesFloor.get(i, j) == FloorType.IceTrap) {
```

Comment:
How about a switch statement instead? Though since we plan to refactor it soon anyway let's leave it as is.